lib/restbase.js

```diff
@@ -307,6 +318,17 @@ RESTBase.prototype._request = function (req) {
             self.metrics.endTiming([statName + statusClass, statName + 'ALL'], startTime);
             throw err;
         });
+        if (match.permissions
+                && Array.isArray(match.permissions)
+                && match.permissions.length > 0) {
+            reqHandlerPromise = reqHandlerPromise.then(function(res) {
```

Comment:
We should actually be able to run the permission checks in parallel with the main request. A good option to do this is something like
 
```javascript
return P.all([reqHandlerPromise, auth.CheckPermissions(..)])
.then(function(res) {
   return res[0];
});
```
A failed permission check will cause P.all to fail as well.